libsession/src/main/java/org/session/libsession/snode/SnodeAPI.kt

```diff
@@ -761,6 +761,62 @@ object SnodeAPI {
         }
     }
 
+    fun updateExpiry(updatedExpiryMs: Long, serverHashes: List<String>): Promise<Map<String, Pair<List<String>, Long>>, Exception> {
+        return retryIfNeeded(maxRetryCount) {
+            val module = MessagingModuleConfiguration.shared
+            val userED25519KeyPair = module.getUserED25519KeyPair() ?: return@retryIfNeeded Promise.ofFail(Error.NoKeyPair)
+            val userPublicKey = module.storage.getUserPublicKey() ?: return@retryIfNeeded Promise.ofFail(Error.NoKeyPair)
+            val updatedExpiryMsWithNetworkOffset = updatedExpiryMs + clockOffset
+            getSingleTargetSnode(userPublicKey).bind { snode ->
+                retryIfNeeded(maxRetryCount) {
+                    // "expire" || expiry || messages[0] || ... || messages[N]
+                    val verificationData =
+                        (Snode.Method.Expire.rawValue + updatedExpiryMsWithNetworkOffset + serverHashes.fold("") { a, v -> a + v }).toByteArray()
+                    val signature = ByteArray(Sign.BYTES)
+                    sodium.cryptoSignDetached(
+                        signature,
+                        verificationData,
+                        verificationData.size.toLong(),
+                        userED25519KeyPair.secretKey.asBytes
+                    )
+                    val params = mapOf(
+                        "pubkey" to userPublicKey,
+                        "pubkey_ed25519" to userED25519KeyPair.publicKey.asHexString,
+                        "expiry" to updatedExpiryMs,
+                        "messages" to serverHashes,
+                        "signature" to Base64.encodeBytes(signature)
+                    )
+                    invoke(Snode.Method.Expire, snode, params, userPublicKey).map { rawResponse ->
+                        val swarms = rawResponse["swarm"] as? Map<String, Any> ?: return@map mapOf()
+                        val result = swarms.mapNotNull { (hexSnodePublicKey, rawJSON) ->
+                            val json = rawJSON as? Map<String, Any> ?: return@mapNotNull null
+                            val isFailed = json["failed"] as? Boolean ?: false
+                            val statusCode = json["code"] as? String
+                            val reason = json["reason"] as? String
+                            hexSnodePublicKey to if (isFailed) {
+                                Log.e("Loki", "Failed to update expiry for: $hexSnodePublicKey due to error: $reason ($statusCode).")
+                                listOf<String>() to 0L
+                            } else {
+                                val hashes = json["updated"] as List<String>
+                                val expiryApplied = json["expiry"] as Long
+                                val signature = json["signature"] as String
```

Comment:
I guess that's an optimisation, but we're still checking that this expiration is sooner than the set expiration (hopefully).

Reply:
ah yep, just meant that we could get the result of the unchanged expiries (and prevent the need for a further `expiries` call) by reading the unchanged values returned here and updating locally from that. Would save us having to schedule any other batch calls to get the expiries or anything down the track

Reply:
I wasn't actually calling this yet. I guess I saw "correct" behaviour after-read because the 1st-device-to-read was probably manually calling delete on the node. So I'm working on adding a call to shorten expiry now.

Reply:
I'm using shorten now.

I don't think utilising `unchanged` would offer significant performance improvements.